cmd/pinshare-tray/tray.go

```diff
@@ -0,0 +1,686 @@
+	t.menuStatus.Disable()
+
+	t.menuIPFSStatus = systray.AddMenuItem("  IPFS: Unknown", "IPFS daemon status")
+	t.menuIPFSStatus.Disable()
+
+	t.menuPinShareStatus = systray.AddMenuItem("  PinShare: Unknown", "PinShare backend status")
+	t.menuPinShareStatus.Disable()
+
+	t.menuPeersStatus = systray.AddMenuItem("  Peers: Unknown", "Connected peers")
+	t.menuPeersStatus.Disable()
+
+	systray.AddSeparator()
+
+	// Service control
+	t.menuStart = systray.AddMenuItem("Start Service", "Start the PinShare service")
+	t.menuStop = systray.AddMenuItem("Stop Service", "Stop the PinShare service")
+	t.menuRestart = systray.AddMenuItem("Restart Service", "Restart the PinShare service")
+
+	systray.AddSeparator()
+
+	// Settings and logs
+	t.menuSettings = systray.AddMenuItem("Settings...", "Open settings")
+	t.menuLogs = systray.AddMenuItem("View Logs...", "Open log directory")
+
+	systray.AddSeparator()
+
+	// About
+	t.menuAbout = systray.AddMenuItem("About PinShare", "About this application")
+
+	systray.AddSeparator()
+
+	// Exit
+	t.menuExit = systray.AddMenuItem("Exit", "Exit the PinShare tray application")
+
+	// Handle menu clicks
+	go t.handleMenuClicks()
+
+	// Initial status check
+	t.updateStatus()
+}
+
+// handleMenuClicks handles menu item clicks
+func (t *Tray) handleMenuClicks() {
+	for {
+		select {
+		// TODO: Re-enable when UI is ready
+		// case <-t.menuOpenUI.ClickedCh:
+		// 	t.handleOpenUI()
+
+		case <-t.menuStart.ClickedCh:
+			t.handleStartService()
+
+		case <-t.menuStop.ClickedCh:
+			t.handleStopService()
+
+		case <-t.menuRestart.ClickedCh:
+			t.handleRestartService()
+
+		case <-t.menuSettings.ClickedCh:
+			t.handleSettings()
+
+		case <-t.menuLogs.ClickedCh:
+			t.handleViewLogs()
+
+		case <-t.menuAbout.ClickedCh:
+			t.handleAbout()
+
+		case <-t.menuExit.ClickedCh:
+			systray.Quit()
+			return
+		}
+	}
+}
+
+// TODO: Re-enable when UI is ready
+// // handleOpenUI opens the PinShare UI in browser
+// func (t *Tray) handleOpenUI() {
+// 	url := fmt.Sprintf("http://localhost:%d", uiPort)
+// 	if err := openBrowser(url); err != nil {
+// 		log.Printf("Failed to open browser: %v", err)
+// 		showMessage("Error", "Failed to open browser")
+// 	}
+// }
+
+// handleStartService starts the service
+func (t *Tray) handleStartService() {
+	if err := startService(); err != nil {
+		log.Printf("Failed to start service: %v", err)
+		showError("PinShare", fmt.Sprintf("Failed to start service:\n\n%v", err))
+	} else {
+		time.Sleep(serviceActionDelay)
+		t.updateStatus()
+	}
+}
+
+// handleStopService stops the service
+func (t *Tray) handleStopService() {
+	if err := stopService(); err != nil {
+		log.Printf("Failed to stop service: %v", err)
+		showError("PinShare", fmt.Sprintf("Failed to stop service:\n\n%v", err))
+	} else {
+		time.Sleep(serviceActionDelay)
+		t.updateStatus()
+	}
+}
+
+// handleRestartService restarts the service
+func (t *Tray) handleRestartService() {
```

Comment:
Is there a reason not to call `handleStartService` and `handleStopService` instead of duplicating their business logic here?